src/ncp/rcp_host.cpp

```diff
@@ -438,12 +451,83 @@ const char *RcpHost::GetThreadVersion(void)
     return version;
 }
 
+static bool areDatasetsEqual(const otOperationalDatasetTlvs &aLhs, const otOperationalDatasetTlvs &aRhs)
+{
+    bool result = false;
+
+    otOperationalDataset     lhsDataset;
+    otOperationalDataset     rhsDataset;
+    otOperationalDatasetTlvs lhsNormalizedTlvs;
+    otOperationalDatasetTlvs rhsNormalizedTlvs;
+
+    // Sort the TLVs in the TLV byte arrays by leveraging the deterministic nature of the two OT APIs
+    SuccessOrExit(otDatasetParseTlvs(&aLhs, &lhsDataset));
```

Comment:
This will drop an TLVs that are unknown to `otDatasetParseTlvs`. Will this cause any issues?

Is it sufficient to simply compare the Network Key and possibly Extended PAN ID?

Reply:
As we discussed offline, this step here is to avoid unnecessary re-join. So it is sufficient to compare network key and extended PAN ID (and we also add channel which may effect connectivity if updated).

> This will drop an TLVs that are unknown to otDatasetParseTlvs. Will this cause any issues?
I added one step to update the new dataset in this if block. So even we decide to skip the re-join, the new dataset will still be set.